src/proxy.sol

```diff
@@ -50,19 +50,26 @@ contract DSProxy is DSAuth, DSNote {
         response = execute(target, _data);
     }
 
+
     function execute(address _target, bytes _data)
         public
         auth
         note
         payable
-        returns (bytes32 response)
+        returns (bytes response)
     {
         require(_target != 0x0);
 
         // call contract in current context
         assembly {
-            let succeeded := delegatecall(sub(gas, 5000), _target, add(_data, 0x20), mload(_data), 0, 32)
-            response := mload(0)      // load delegatecall output
+            let succeeded := delegatecall(sub(gas, 5000), _target, add(_data, 0x20), mload(_data), 0, 0)
+            let size := returndatasize
+
+            response := mload(0x40)
+            mstore(0x40, add(response, and(add(add(size, 0x20), 0x1f), not(0x1f))))
```

Comment:
What is special about `0x1f`?

Reply:
Actually did not get through deeply on that but is how it shows in the documentation how you set up the new memory end including a padding: https://solidity.readthedocs.io/en/v0.4.25/assembly.html#example

Reply:
Other option would be doing something like this:
```
function execute(address _target, bytes _data)
        public
        auth
        note
        payable
        returns (bytes response)
    {
        require(_target != 0x0);

        // call contract in current context
        uint size;
        bool succeeded;
        assembly {
            succeeded := delegatecall(sub(gas, 5000), _target, add(_data, 0x20), mload(_data), 0, 0)
            size := returndatasize
            response := mload(0x40)
        }
        response = new bytes(size);         
        assembly {
            mstore(response, size)
            returndatacopy(add(response, 0x20), 0, size)

            switch iszero(succeeded)
            case 1 {
                // throw if delegatecall failed
                revert(0, 0)
            }
        }
    }
```

Reply:
I'm not really sure why the padding is necessary.

Reply:
@rainbreak I think I know why this math is applied. If I'm not wrong `size` could be any value.
If size is multiple of 32, then `and(add(add(size, 0x20), 0x1f), not(0x1f))` is the same than just doing `add(size, 0x20)`.
But if `size` is between two 32 bytes multiple numbers, this math will make the value to match the higher one.

Reply:
the higher one + 32

Reply:
Actually I'm not sure if exists a case when size is not multiple of 32. Solidity packs it always as 32 bytes, so wondering why the extra check.

Reply:
What if you aren't calling a contract compiled with Solidity?

Reply:
actually using assembly you can return a not 32 bytes multiple value

Reply:
I think for this specific case, we will always have the exact same result regardless having the padding or not. It doesn't matter if you put the end of memory pointer at size space or to the next 32 multiple number. It will only return the response value set to the size of the delegatecall result.
Maybe not leaving the padding could provoke some issues if the code would continue and more variables would be processed and returned. I should have to do more testing to figure it out if there would be some impact.
I guess it is a good practice to keep the padding, but if you feel it shouldn't be there I can remove.

Reply:
Yeah that would make sense. We should have a test for this.